### ISSUE TITLE
Exposing errors in the message field would greatly improve debugging

### ISSUE THREAD

**Issue by jorhett** [hashicorp/boundary#795]
**Is your feature request related to a problem? Please describe.**

Errors regarding scope or duplicate assignment give 500 errors with no detail:

```
$ boundary users add-accounts -id=u_BarFoo1234 -account=apw_FooBar1234
Error from controller when performing add-accounts on user:
Error information:
  Code:                Internal
  Error ID:            1B7vyMQ0bD
  Message:
  Status:              500
```

**Describe the solution you'd like**

The log information available via `journalctl` is very informative. If this could be returned in the *Message* field it would greatly assist debugging. 

```
Nov 18 23:44:29 boundary-worker-01 boundary[41692]: 2020-11-18T23:44:29.946Z [ERROR] controller: internal error returned: error id=zJClPrNyng error="Status: 500, Code: "Internal", Error: "Unable to set accounts for the user: set associated accounts: set associated accounts: unable to associate ids: associate user with accounts: apw_FooBar1234 account is associated with a user u_BarFoo1234: invalid parameter.""
Nov 18 23:46:25 boundary-worker-01 boundary[41692]: 2020-11-18T23:46:25.956Z [ERROR] controller: internal error returned: error id=1B7vyMQ0bD error="Status: 500, Code: "Internal", Error: "Unable to add accounts to user: associate accounts: error associating account ids: associate accounts: associate user with accouts: failed to associate apw_FooBar1234 account: update: failed: pq: new row for relation \"auth_account\" violates check constraint \"user_and_auth_account_in_same_scope\".""
```

**Describe alternatives you've considered**

none

**Explain any additional use-cases**

Helps return useful responses to the user/admin.

**Additional context**

none

**Comment by malnick** [comment]
Thanks for submitting this @jorhett.

We're in the middle of a big refactor of how we do error handling but we're still finalizing how to do this responsibly without leaking sensitive information to arbitrary clients. More to come, please stay tuned!

**Comment by louisruch** [comment]
This specific issue was resolved with #784 and #841:

```
boundary users add-accounts -id=u_anon -account=apw_XrXOEaMm2v
Error from controller when performing add-accounts on user: 
Error information:
  Kind:                Internal
  Message:             Unable to add accounts to user: iam.(Repository).AddUserAccounts: db.DoTx: iam.(Repository).AddUserAccounts: iam.associateUserWithAccounts: apw_XrXOEaMm2v account is associated with a user u_1234567890: parameter violation: error #100.
  Status:              500
```

As we refactor the remaining repos other commands will also return the required information.
